### PR TITLE
feat(PocketIC): specify replica log level of PocketIC instances

### PR DESCRIPTION
**Pull request by mraszyk**
This PR adds support for specifying replica log level of PocketIC instances.

### CODE REVIEW

File: rs/pocket_ic_server/src/pocket_ic.rs [hunk 1]
```diff
@@ -163,6 +163,7 @@ pub struct PocketIc {
     registry_data_provider: Arc<ProtoRegistryDataProvider>,
     runtime: Arc<Runtime>,
     nonmainnet_features: bool,
+    log_level: Option<String>,
```
Comment (schneiderstefan) [inline rs/pocket_ic_server/src/pocket_ic.rs:166]: Wouldn't it make more sense if this was type `Level`?

Comment (mraszyk) [reply rs/pocket_ic_server/src/pocket_ic.rs]: Makes sense! Done.

**Review by schneiderstefan — APPROVED** [review]

**Comment by mraszyk** [comment]
> Should we retire this variable?

I'm not sure since the logs produced by the server itself as well as the canister logs are not affected by the log level of an instance (affecting the replica logs of the instance's state machines). In other words, if you want to mute everything, you need to set POCKET_IC_MUTE_SERVER. Also POCKET_IC_MUTE_SERVER is for the PocketIC library only whereas this PR allows users to specify the log level on the server side.

> Also, maybe this is a good point in time to mention how to see/mute log outputs with PocketIC in some readme (maybe HOWTO.md)

Agreed.

**Comment by mraszyk** [comment]
@fxgst I've added a note on using `POCKET_IC_MUTE_SERVER` in HOWTO.md.

**Comment by michael-weigelt** [comment]
Why is it desirable to set log level per instance?

**Comment by mraszyk** [comment]
> Why is it desirable to set log level per instance?

Would you prefer setting a global log level for the entire PocketIC server?

**Comment by mraszyk** [comment]
> I'd love it to be easy to switch between:
> 
> * logging to stdout at a specified level, by default set to Error (so you see very little output)
> * logging to a file at a specified location (perhaps this should always happen, even if no file given, e.g. logfile to /tmp with launch timestamp in name)
> 
> The logfile could be handy when people have a problem and can just send us their log files. Or on CI runs etc.

I see, but this is orthogonal to the log level. My suggestion would be to follow the [logger](https://github.com/dfinity/ic/blob/234ca3809b53192743763c0f7607e00da0ea2dce/rs/config/src/logger.rs#L39) configuration in the replica: specify the log level (this MR) and the log destination (follow up MR) separately. Is this sufficient or do you need multiple log stream with their log levels and destinations?

**Comment by mraszyk** [comment]
Also note that separating the log streams per instance (test) has been requested by @aterga so adding a log destination per instance (test) in a follow up PR might also address the NNS team's use case.

**Review by michael-weigelt — APPROVED** [review]

**Pull request merged**